clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -42,20 +41,26 @@
     private final int serializedValueSize;
     private final TopicPartition topicPartition;
 
-    private volatile Long checksum;
-
     public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset, long timestamp,
-                          Long checksum, int serializedKeySize, int serializedValueSize) {
+                          int serializedKeySize, int serializedValueSize) {
         // ignore the relativeOffset if the base offset is -1,
         // since this indicates the offset is unknown
         this.offset = baseOffset == -1 ? baseOffset : baseOffset + relativeOffset;
         this.timestamp = timestamp;
-        this.checksum = checksum;
         this.serializedKeySize = serializedKeySize;
         this.serializedValueSize = serializedValueSize;
         this.topicPartition = topicPartition;
     }
 
+    /**
+     * @deprecated use constructor without `checksum` parameter.
+     */
+    @Deprecated
+    public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset, long timestamp,
```

Comment:
Not sure whether we have to offer this constructor. We remove the `checksum` from all constructors of `ConsumerRecord` even though we don't add deprecation annotation to  constructors before.

Reply:
I probably need to add back some of the `ConsumerRecord` constructors since they were not previously deprecated.